daemon/src/position_metrics.rs

```diff
@@ -322,4 +354,94 @@ mod metrics {
         cfds.iter()
             .fold(Usd::ZERO, |sum, cfd| cfd.quantity_usd + sum)
     }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+
+        fn create_cfd(
+            is_open: bool,
+            is_closed: bool,
+            is_failed: bool,
+            is_refunded: bool,
+            is_rejected: bool,
+        ) -> Cfd {
+            Cfd {
+                id: Default::default(),
+                position: Position::Long,
+                quantity_usd: Usd::ZERO,
+                is_open,
+                is_closed,
+                is_failed,
+                is_refunded,
+                is_rejected,
+                version: 0,
+            }
+        }
+
+        #[test]
+        fn when_cfd_is_in_single_no_state_then_is_not_unknown() {
+            let state_matrix = [
+                [false, false, false, false, false],
+                [true, false, false, false, false],
+                [false, true, false, false, false],
+                [false, false, true, false, false],
+                [false, false, false, true, false],
+                [false, false, false, false, true],
+            ];
+
+            for states in state_matrix {
+                let is_open = states[0];
+                let is_closed = states[1];
+                let is_failed = states[2];
+                let is_refunded = states[3];
+                let is_rejected = states[4];
+                let cfd = create_cfd(is_open, is_closed, is_failed, is_refunded, is_rejected);
```

Comment:
I think this would be more readable if you would define a (test-only) struct which has 5 boolean fields that you can use to initialize your `Cfd` aggregate with:

```rust
struct StateFlags {
	open: bool,
	closed: bool,
	failed: bool,
	refunded: bool,
	rejected: bool
}
```

Then your state matrix definition can be:

```rust
let state_matrix = [
	StateFlags { open: false, closed: false, failed: false, refunded: false, rejected: false },
	// ...
];
```

Reply:
You could even do things like:

```rust
StateFlags {
	open: true,
	..StateFlags::all_false()
}
```

Reply:
I just tried it but I think it is easier with a matrix to see that all combinations are covered. I'll do a mix.